diff -ruN a/lposix.c b/lposix.c
--- a/lposix.c	2008-07-18 09:30:36.000000000 -0500
+++ b/lposix.c	2010-06-24 12:49:23.000000000 -0500
@@ -575,6 +575,12 @@
 	return 1;
 }
 
+static int Pusleep(lua_State *L)			/** usleep(usec) */
+{
+	useconds_t usec = luaL_checklong(L, 1);
+	return pushresult(L, usleep(usec), NULL);
+}
+
 
 static int Psetenv(lua_State *L)		/** setenv(name,value,[over]) */
 {
@@ -1069,6 +1075,7 @@
 	{"setenv",		Psetenv},
 	{"setpid",		Psetpid},
 	{"sleep",		Psleep},
+	{"usleep",		Pusleep},
 	{"stat",		Pstat},
 	{"sysconf",		Psysconf},
 	{"times",		Ptimes},
diff -ruN a/lposix.h b/lposix.h
--- a/lposix.h	1969-12-31 18:00:00.000000000 -0600
+++ b/lposix.h	2010-06-22 15:29:02.000000000 -0500
@@ -0,0 +1,6 @@
+#ifndef LPOSIX_H
+#define LPOSIX_H
+
+LUALIB_API int luaopen_posix (lua_State *L);
+
+#endif
diff -ruN a/Makefile b/Makefile
--- a/Makefile	2008-07-18 09:31:56.000000000 -0500
+++ b/Makefile	2010-06-22 15:29:02.000000000 -0500
@@ -25,7 +25,7 @@
 
 CPPFLAGS=	-fPIC $(INCS) $(WARN)
 WARN=		-pedantic -Wall
-INCS=		-I$(LUAINC)
+INCS=		`pkg-config --cflags lua5.1 2> /dev/null || pkg-config --cflags lua 2> /dev/null`
 
 MYNAME=		posix
 MYLIB= 		$(MYNAME)
